kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/FilterHarness.java

```diff
@@ -86,16 +87,20 @@ protected void buildChannel(Filter... filters) {
         var inboundChannel = new EmbeddedChannel();
         var channelProcessors = Stream.<ChannelHandler> of(new InternalRequestTracker(), new CorrelationIdIssuer());
 
-        clientSaslManager = new ClientSaslManager();
         ProxyChannelStateMachine channelStateMachine = new ProxyChannelStateMachine(testVirtualCluster.getClusterName(), null);
+
+        clientSaslManager = ClientSaslManager
+                .create(inboundChannel)
+                .bindManager(inboundChannel, Subject.anonymous());
```

Comment:
Why do we need to supply `inboundChannel` on `bindManager` and `create`. I would expect `create` to create an instance that was implicitly connected to `inboundChannel`. Is there a use case for create without connecting the result to the channel?